src/Log/Engine/FileLog.php

```diff
@@ -128,13 +137,17 @@ public function log($level, $message, array $context = []): void
         $pathname = $this->_path . $filename;
         $mask = $this->_config['mask'];
         if (!$mask) {
-            file_put_contents($pathname, $output, FILE_APPEND);
+            foreach ($formatted as $output) {
+                file_put_contents($pathname, $output . "\n", FILE_APPEND);
```

Comment:
Not sure if it's cheaper to append a newline which creates a new string or call file_put_contents again with just `"\n"`.

Reply:
This looks good to me like this.